### PR TITLE
ATB-1139 | Adding TTL to Core Template

### PR DESCRIPTION
**Pull request by jessica-li-01**
## Proposed changes

[ATB-1139]

### What changed

- Switch on TTL configurations on account status table
### Why did it change

- To add TTL to the table for better tracking

### Issue tracking

- [ATB-1139](https://govukverify.atlassian.net/browse/ATB-1139)

## Testing

TTL comes up as option:
Screenshot 2023-11-15 at 14 23 35
Screenshot 2023-11-15 at 14 23 11
Screenshot 2023-11-15 at 14 22 51
Screenshot 2023-11-15 at 14 38 30

## Checklists
- [ ] Did not commit any not-required changes to the `src/infra/**/samconfig.toml`
- [ ] Tested changes and included test evidence in the PR, if appropriate
- [ ] Included all required tags and other properties for any new resources in the SAM template
- [ ] Ensured that any new resources in the SAM Template follow appropriate naming conventions
- [ ] Ensured that naming of new resources is compatible with deploying multiple stacks with custom stack names during development
- [ ] Ensured that no log lines include PII or other sensitive data
- [ ] Implemented unit testing for any new logic implemented, if appropriate
- [ ] Ensured that all commits in this PR are signed
- [ ] Ensured appropriate code coverage is maintained by unit tests

[ATB-1139]: https://govukverify.atlassian.net/browse/ATB-1139?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ATB-1139]: https://govukverify.atlassian.net/browse/ATB-1139?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TaherKapasi** [comment]
Hi @jessica-li-01 - can you test this by writing a record with a time to live set a minute or so into the future (get epoch time in seconds and write it into the table), then wait until its automatically deleted - could you include screenshots of before and after -thanks

File: src/services/dynamo-database-service.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export class DynamoDatabaseService {
       return response;
     } catch (error: any) {
       if (!error.name || error.name !== 'ConditionalCheckFailedException') {
-        logger.error(`${LOGS_PREFIX_SENSITIVE_INFO} Error updating item with pk ${userId}.`);
+        logger.error(`${LOGS_PREFIX_SENSITIVE_INFO} Error updating Dynamo DB ${error}.`);
```
Comment (TaherKapasi) [inline src/services/dynamo-database-service.ts:106]: could we print the message and include error as a object

**Review by TaherKapasi — CHANGES_REQUESTED** [review]

File: src/services/dynamo-database-service.ts (hunk 1 above)
Comment (jessica-li-01) [reply src/services/dynamo-database-service.ts:106]: It doesn't like it because the error is of type unknown/any

Comment (TaherKapasi) [reply src/services/dynamo-database-service.ts]: Hi @jessica-li-01 - see latest commit

**Review by chiarapascucci-gds — APPROVED** [review]

**Pull request merged**